test/jdk/javax/swing/ImageIcon/ImageIconTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8159055
+ * @summary Verifies ImageIcon.setImage handles null parameter
+ * @run main ImageIconTest
+ */
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.net.URL;
+import java.util.Random;
+import java.awt.Image;
+import java.awt.Toolkit;
+import javax.swing.ImageIcon;
+
+public class ImageIconTest {
+
+    static enum ArgType { FILE, URL, BYTE_ARRAY, IMAGE, SET_IMAGE };
+    static enum ArgVal { NULL, INVALID_DATA };
+
+    public static void main(String[] args) throws Exception {
+
+        String imgName = "invalid.gif";
+        byte[] invalidData = new byte[100];
+        new Random().nextBytes(invalidData);
+        try (FileOutputStream fos = new FileOutputStream(imgName)) {
+            fos.write(invalidData);
+        }
+        String fileName = (new File(System.getProperty("test.src", "."), imgName)).getName();
```

Comment:
Why do we need jumping over the hoops?

You're creating an image, put it into the current directory that defaults to `scratch` directory in jtreg. The `scratch` directory is automatically cleaned by jtreg between tests, thus you won't even leave files in the source tree of JDK.

Reply:
used file.deleteOnExit